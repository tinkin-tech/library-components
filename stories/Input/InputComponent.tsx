import * as React from 'react'
import language from './language/es_EC'

export type IInputTypes = 'text' | 'email' | 'number' | 'password'

export interface InputComponentPropsInterface {
  valueId: string
  value: string
  onChangeValue: (value: string, valueId: string) => void
  type: IInputTypes
  label?: string
  required?: boolean
  placeholder?: string
  error?: string
  labelClassName?: string
  inputClassName?: string
  readOnly?: boolean
  textArea?: boolean
}

const InputComponent: React.FC<InputComponentPropsInterface> = (
  props: InputComponentPropsInterface
) => {
  const {
    value,
    valueId,
    onChangeValue,
    type,
    label,
    required,
    error,
    placeholder,
    labelClassName,
    inputClassName,
    readOnly,
    textArea,
  } = props

  const onChangeAction = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void => {
    let currentValue = event.currentTarget.value
    if (type === 'number') {
      currentValue = currentValue.replace(/[^\d.-]/g, '')
    }
    onChangeValue(currentValue, valueId)
  }

  const propsComponent = {
    id: valueId,
    name: valueId,
    value: value || '',
    autoComplete: 'off',
    spellCheck: false,
    onChange: readOnly ? null : onChangeAction,
    type: type === 'number' ? 'text' : type,
    placeholder: placeholder || language.placeholder,
    className: `${inputClassName || ''} ${error ? 'warning' : ''}`,
    disabled: readOnly,
  }

  return (
    <div className={`input-component${readOnly ? ' disabled' : ''}`}>
      <div className="flex-space-between">
        <label
          className={`${labelClassName || 'label'} ${error ? 'warning' : ''}`}
          htmlFor={valueId}
          data-testid="label-component"
        >
          {`${label || ''}${required ? '*' : ''}`}&nbsp;
        </label>
        {error && <span className="warning-content">&#9888;</span>}
      </div>
      {textArea ? (
        <textarea {...propsComponent} />
      ) : (
        <input {...propsComponent} />
      )}
      {error && <span className="error-message">{error}</span>}
    </div>
  )
}

export default InputComponent
