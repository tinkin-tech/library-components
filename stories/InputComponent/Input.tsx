import * as React from 'react'
import { LabelComponent } from '../label/Label'

export type IInputTypes = 'text' | 'email' | 'number' | 'password'

export interface InputComponentPropsInterface {
  id: string
  value: string
  onChangeValue: (id: string, value: string | number) => void
  type: IInputTypes
  label?: string
  required?: boolean
  placeholder?: string
  error?: string
  disabled?: boolean
}

export const InputComponent: React.FC<InputComponentPropsInterface> = (
  props: InputComponentPropsInterface
) => {
  const {
    value,
    id,
    onChangeValue,
    type,
    label,
    required,
    error,
    disabled,
    placeholder,
  } = props

  const [valueState, changeValueState] = React.useState(value)

  const onChangeAction = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void => {
    changeValueState(event.currentTarget.value)
    onChangeValue(id, event.currentTarget.value)
  }

  React.useEffect(() => {
    if (value !== valueState) {
      changeValueState(value)
    }
  }, [value])

  return (
    <div
      id="input-component"
      className="input-component input p-t p-b flex-column"
    >
      <div className="flex-space-between">
        <LabelComponent
          referenceId={id}
          label={label}
          error={!!error}
          required={required}
          disabled={disabled}
        />
        {error && <span className="icon-warning-content">&#9888;</span>}
      </div>
      <input
        id={id}
        name={id}
        data-testid="input-component"
        value={valueState}
        autoComplete="off"
        spellCheck={false}
        onChange={onChangeAction}
        type={type}
        placeholder={placeholder}
        disabled={disabled}
        className={error ? `warning` : ''}
      />
      {error && <div className="error-message">{error}</div>}
    </div>
  )
}
