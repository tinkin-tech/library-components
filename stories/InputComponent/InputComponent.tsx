import * as React from 'react'

export type IInputTypes = 'text' | 'email' | 'number' | 'password'

export type ILabelPositionTypes = 'outside' | 'inside'

export type IErrorPositionTypes = 'left' | 'right'

export interface InputComponentPropsInterface {
  /**
   * Id del InputComponent
   */
  id: string
  /**
   * Valor del InputComponent
   */
  value: string
  /**
   * Función que recibe (id, value) cuando se realiza un cambio en la
   * entrada(InputComponent)
   */
  onChangeValue: (id: string, value: string | number) => void
  /**
   * Tipo del InputComponent
   */
  type: IInputTypes
  /**
   * Etiqueta(label) del InputComponent
   */
  label?: string
  /**
   * La posición de la etiqueta(label) del InputComponent
   */
  labelPosition?: ILabelPositionTypes
  /**
   * Se representa con un asterisco si la entrada(InputComponent) es requerido
   */
  required?: boolean
  /**
   * Texto mostrado cuando no hay valor en la entrada(InputComponent)
   */
  placeholder?: string
  /**
   *  Valor que representa un error. Al ingresar un valor un estilo error
   *  se muestra en la entrada, junto al valor que ingrese en la propiedad
   *  error
   */
  error?: string
  /**
   * La posición del error del InputComponent
   */
  errorPosition?: IErrorPositionTypes
  /**
   * Cuando el valor es verdadero, muestra un estilo opaco en el
   * componente y el valor no se puede ingresar en la entrada.
   */
  disable?: boolean
}

type errorsLocalTypes = 'ONLY_NUMBERS' | 'FORMAT_EMAIL'

type errorsLocalEnumsTypes = {
  [name in errorsLocalTypes]: {
    key: errorsLocalTypes
    label: string
  }
}

const ERRORS_LOCAL_ENUMS: errorsLocalEnumsTypes = {
  ONLY_NUMBERS: {
    key: 'ONLY_NUMBERS',
    label: 'Ingrese solo números y [-, .]',
  },
  FORMAT_EMAIL: {
    key: 'FORMAT_EMAIL',
    label: 'Ingrese un formato válido de correo',
  },
}

/**
 * Componente UI para interacción
 */
export const InputComponent: React.FC<InputComponentPropsInterface> = (
  props: InputComponentPropsInterface
) => {
  const {
    value,
    id,
    onChangeValue,
    type,
    label,
    labelPosition,
    required,
    error,
    disable,
    placeholder,
    errorPosition,
  } = props

  const [localError, changeLocalError] = React.useState('')

  const filterValueForType = (value: string | number): string | number => {
    const valueString = value.toString()
    switch (type) {
      case 'number': {
        const reg = new RegExp(/[^\d.-]/g)
        value &&
          changeLocalError(
            valueString.split('').some((element) => reg.test(element))
              ? ERRORS_LOCAL_ENUMS.ONLY_NUMBERS.label
              : ''
          )
        return value ? value.toString().replace(reg, '') : ''
      }
      case 'email': {
        const reg = new RegExp(/^[a-zA-Z0-9_.\-]+@[a-z]+\.[a-z]+$/)
        value &&
          changeLocalError(
            reg.test(valueString) ? '' : ERRORS_LOCAL_ENUMS.FORMAT_EMAIL.label
          )
        return value
      }
      default: {
        return value
      }
    }
  }

  const [localValue, changeLocalValue] = React.useState(
    filterValueForType(value)
  )

  const onChangeAction = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void => {
    const newValue = event.currentTarget.value
    changeLocalValue(filterValueForType(newValue))
    onChangeValue(id, filterValueForType(newValue))
  }

  const onClickInputContent = (): void => {
    if (!disable) {
      // eslint-disable-next-line no-restricted-globals
      document.getElementById(id).focus()
    }
  }

  React.useEffect(() => {
    if (value !== localValue) {
      changeLocalValue(filterValueForType(value))
    }
  }, [value])

  const labelElement = label && (
    <label
      className={`label ${error || localError ? 'label-error' : ''}`}
      htmlFor={id}
    >
      {`${label}${required ? '*' : ''}`}
    </label>
  )

  return (
    <div
      id="input-component"
      className={`input-component input flex-column ${
        disable ? 'disable-component' : ''
      }`}
    >
      <label htmlFor={id}>{label && labelElement}</label>
      {label && labelPosition === 'inside' && labelElement}
      <input
        id={id}
        name={id}
        data-testid="input-component"
        value={localValue}
        autoComplete="off"
        spellCheck={false}
        onChange={onChangeAction}
        type={type !== 'number' ? type : 'text'}
        placeholder={placeholder}
        disabled={disable}
        className={error || localError ? `warning` : ''}
      />
      {(error || localError) && (
        <div className="icon-warning-content">
          <div className="i-warning" />
        </div>
      )}
      {(error || localError) && (
        <div
          className={`error-message ${
            errorPosition === 'right' ? 'error-position-right' : ''
          }`}
        >
          {error || localError}
        </div>
      )}
    </div>
  )
}
