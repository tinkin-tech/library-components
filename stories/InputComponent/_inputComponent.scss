//$border-color: $color-primary;
//$focus-color: #72A6FF;
//$error-color: #DA4900;
//$background-color: #FAFAFA;
//$background-focus-color: #ECF3FF;
//$disable-color: #C4C4C4;
//$input-color: #2A2A2A;
//
//$input-component-border: 1px solid $border-color;
//$input-component-border-color-focus: $focus-color;
//$input-component-border-color-error: $error-color;
//$input-component-border-radius: 4px;
//$input-component-width: 100%;
//$input-component-background-color: $background-color;
//$input-component-background-color-focus: $background-focus-color;
//$input-component-padding: 2px 8px;
//$label-padding-top: 4px;
//$label-size: 12px;
//$label-color: $border-color;
//$label-error-color: $error-color;
//$label-error-padding-right: 30px;
//$label-color-focus: $focus-color;
//$disable-text-color: $disable-color;
//$disable-text-padding-without-label: 27px 0 0 0;
//$input-padding: 4px 0;
//$input-font-size: 14px;
//$input-padding-right-error: 24px;
//$input-width: calc(100% - 16px);
//$icon-top: 3px;
//$icon-right: 8px;
//$error-size: 10px;
//$error-padding: 2px;
//
//.input-component{
//  width: $input-component-width;
//  font-family: muli, sans-serif;
//  &:focus-within {
//    .label {
//      @include transition(all 0.5s);
//      color: $label-color-focus;
//    }
//  }
//  &.disable-component {
//    opacity: 0.6;
//  }
//  .label {
//    font-size: $label-size;
//    display: block;
//    cursor: pointer;
//    font-weight: bold;
//    color: $label-color;
//    padding-top: $label-padding-top;
//    @include flexBox();
//    @include flex-direction(column);
//    flex: 1;
//    font-family: $font-family;
//    word-break: break-all;
//    content: attr(data-error);
//    @include transition(all 0.5s);
//    &.label-error {
//      @include transition(all 0.5s);
//      color: $label-error-color!important;
//      padding-right: $label-error-padding-right;
//    }
//  }
//  .input-component-content {
//    border: $input-component-border;
//    border-radius: $input-component-border-radius;
//    background-color: $input-component-background-color;
//    width: $input-component-width;
//    position: relative;
//    padding: $input-component-padding;
//    box-sizing: border-box;
//    cursor: pointer;
//    @include transition(all 0.5s);
//    &.input-component-error {
//      @include transition(all 0.5s);
//      border-color: $input-component-border-color-error!important;
//    }
//    &:focus-within {
//      @include transition(all 0.5s);
//      border-color: $input-component-border-color-focus;
//      background-color: $input-component-background-color-focus;
//      .label {
//        @include transition(all 0.5s);
//        color: $label-color-focus;
//      }
//    }
//    .disable-text {
//      padding: $input-padding;
//      color: $disable-text-color;
//      font-family: $font-family;
//      font-size: $input-font-size;
//      &.disable-text-without-label {
//        padding: $disable-text-padding-without-label;
//      }
//    }
//    input {
//      border: none;
//      background-color: transparent;
//      padding: $input-padding;
//      font-size: $input-font-size;
//      text-overflow: ellipsis;
//      white-space: nowrap;
//      width: $input-width;
//      color: $input-color;
//      font-family: muli, sans-serif;
//      &::placeholder {
//        color: $disable-text-color;
//      }
//      &:focus {
//        outline: none;
//      }
//    }
//    input[type=number], ::-webkit-inner-spin-button {
//      -webkit-appearance: none;
//      -moz-appearance: textfield;
//    }
//    .icon-warning-content {
//      position: absolute;
//      top: $icon-top;
//      right: $icon-right;
//    }
//  }
//  .error-message {
//    font-size: $error-size;
//    font-weight: bold;
//    color: $label-error-color!important;
//    padding: $error-padding;
//    font-family: $font-family;
//    word-break: break-all;
//    &.error-position-right {
//      text-align: right;
//    }
//  }
//}
