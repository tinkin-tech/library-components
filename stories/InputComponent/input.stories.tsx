import * as React from 'react'
import { action } from '@storybook/addon-actions'

import { InputComponent, InputComponentPropsInterface } from './Input'

export default {
  title: 'Input',
  component: InputComponent,
}

const Template = (arg: InputComponentPropsInterface): JSX.Element => (
  <InputComponent {...arg} />
)

export const Default = Template.bind({})

Default.args = {
  /**
   * Id del InputComponent
   */
  id: 'input',
  /**
   * Valor del InputComponent
   */
  value: '',
  /**
   * Funci√≥n que recibe (id, value) cuando se realiza un cambio en la
   * entrada(InputComponent)
   */
  onChangeValue: action('OnChangeValue'),
  /**
   * Tipo del InputComponent
   */
  type: 'text',
  /**
   * Se representa con un asterisco si la entrada(InputComponent) es requerido
   */
  required: false,
  /**
   * Texto mostrado cuando no hay valor en la entrada(InputComponent)
   */
  placeholder: '',
  /**
   *  Valor que representa un error. Al ingresar un valor un estilo error
   *  se muestra en la entrada, junto al valor que ingrese en la propiedad
   *  error
   */
  error: '',
  /**
   * Cuando el valor es verdadero, muestra un estilo opaco en el
   * componente y el valor no se puede ingresar en la entrada.
   */
  disable: false,
}
