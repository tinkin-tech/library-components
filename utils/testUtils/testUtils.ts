export class TestUtil {
  private value: string

  mockedFunction = (value: string, _: string): void => {
    this.setValue(value)
  }

  getValue = (): string => this.value

  private setValue = (value: string): void => {
    this.value = value
  }
}

export const setLocationSearchValue = (search: string): void => {
  const oldWindowLocation = window.location
  delete window.location

  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  window.location = Object.defineProperties(
    {},
    {
      ...Object.getOwnPropertyDescriptors(oldWindowLocation),
      search: {
        configurable: true,
        value: search,
      },
    }
  )
}

/* istanbul ignore next */
export const mockOffsetAndClientSizes = (value?: number): void => {
  const defaultValue = 500

  const originalClientHeight = Object.getOwnPropertyDescriptor(
    Element.prototype,
    'clientHeight'
  )
  const originalClientWidth = Object.getOwnPropertyDescriptor(
    Element.prototype,
    'clientWidth'
  )
  const originalOffsetWidth = Object.getOwnPropertyDescriptor(
    HTMLElement.prototype,
    'offsetWidth'
  )
  const originalOffsetHeight = Object.getOwnPropertyDescriptor(
    HTMLElement.prototype,
    'offsetHeight'
  )
  beforeAll(() => {
    Object.defineProperty(Element.prototype, 'clientHeight', {
      configurable: true,
      value: value || defaultValue,
    })
    Object.defineProperty(Element.prototype, 'clientWidth', {
      configurable: true,
      value: value || defaultValue,
    })
    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', {
      configurable: true,
      value: value || defaultValue,
    })
    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', {
      configurable: true,
      value: value || defaultValue,
    })
  })

  afterAll(() => {
    Object.defineProperty(
      HTMLElement.prototype,
      'clientHeight',
      originalClientHeight
    )
    Object.defineProperty(
      HTMLElement.prototype,
      'clientWidth',
      originalClientWidth
    )
    Object.defineProperty(
      HTMLElement.prototype,
      'offsetWidth',
      originalOffsetHeight
    )
    Object.defineProperty(
      HTMLElement.prototype,
      'offsetHeight',
      originalOffsetWidth
    )
  })
}
