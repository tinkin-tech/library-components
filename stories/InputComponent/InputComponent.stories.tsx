import * as React from 'react'
import { action } from '@storybook/addon-actions'

import { InputComponent, InputComponentPropsInterface } from './InputComponent'

export default {
  title: 'Input',
  component: InputComponent,
}

const Template = (arg: InputComponentPropsInterface): JSX.Element => (
  <InputComponent {...arg} />
)

export const Primary = Template.bind({})

Primary.args = {
  id: 'input',
  value: '',
  onChangeValue: action('OnChangeValue'),
  type: 'text',
  placeholder: '',
  label: '',
  labelPosition: '',
  required: true,
  error: '',
  errorPosition: '',
  disable: false,
}
