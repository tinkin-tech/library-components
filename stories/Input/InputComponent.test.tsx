import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import InputComponent from './InputComponent'
import { TestUtil } from '../../utils/testUtils/testUtils'

describe('InputComponent test', () => {
  const mockOnChangeInput = jest.fn()

  describe('Value prop', () => {
    it('Should recive value, set in input value', () => {
      const { container } = render(
        <InputComponent
          valueId="input"
          value="generic input value"
          onChangeValue={mockOnChangeInput}
          type="text"
        />
      )
      expect(container.getElementsByTagName('input')[0].value).toBe(
        'generic input value'
      )
    })
  })

  describe('onChangeInput prop', () => {
    it(
      'Should onChangeInput called with input value and valueId, when change ' +
        'input value',
      () => {
        const { container } = render(
          <InputComponent
            valueId="input"
            value=""
            onChangeValue={mockOnChangeInput}
            type="text"
          />
        )
        fireEvent.change(container.getElementsByTagName('input')[0], {
          target: {
            value: 'input value',
          },
        })
        expect(mockOnChangeInput).toHaveBeenCalledTimes(1)
        expect(mockOnChangeInput).toHaveBeenCalledWith('input value', 'input')
      }
    )

    it('Should change input value', () => {
      const mockUtil = new TestUtil()
      const { container, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockUtil.mockedFunction}
          type="text"
        />
      )
      expect(container.getElementsByTagName('input')[0]).toMatchObject({
        value: '',
      })
      fireEvent.change(container.getElementsByTagName('input')[0], {
        target: {
          value: 'hello world',
        },
      })
      rerender(
        <InputComponent
          valueId="input"
          value={mockUtil.getValue()}
          onChangeValue={mockUtil.mockedFunction}
          type="text"
        />
      )
      expect(container.getElementsByTagName('input')[0]).toMatchObject({
        value: 'hello world',
      })
    })
  })

  describe('Type prop', () => {
    it('Should change value if enter only numbers, when type is number', () => {
      const mockUtil = new TestUtil()
      const { container, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockUtil.mockedFunction}
          type="number"
        />
      )
      fireEvent.change(container.getElementsByTagName('input')[0], {
        target: {
          value: 'uhtn123',
        },
      })
      rerender(
        <InputComponent
          valueId="input"
          value={mockUtil.getValue()}
          onChangeValue={mockUtil.mockedFunction}
          type="number"
        />
      )
      expect(container.getElementsByTagName('input')[0]).toMatchObject({
        value: '123',
      })
    })
  })

  describe('label prop', () => {
    it('Should recive label prop and show in component', () => {
      const { getByText } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          label="label input"
        />
      )
      expect(getByText('label input')).toBeInTheDocument()
    })
  })

  describe('required prop', () => {
    it('Should recive required prop and add "*" in label text', () => {
      const { getByText, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          label="My input"
        />
      )
      expect(getByText('My input').textContent).not.toContain('*')
      rerender(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          label="My input"
          required={true}
        />
      )
      expect(getByText('My input*')).toBeInTheDocument()
    })
  })

  describe('placeholder prop', () => {
    it('Should recive placeholder prop and set in placeholder input', () => {
      const { container } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          placeholder="ej: input"
        />
      )
      expect(container.getElementsByTagName('input')[0].placeholder).toBe(
        'ej: input'
      )
    })

    it(
      'Should set in placeholder default value when not recive placeholder' +
        'prop',
      () => {
        const { container } = render(
          <InputComponent
            valueId="input"
            value=""
            onChangeValue={mockOnChangeInput}
            type="text"
          />
        )
        expect(container.getElementsByTagName('input')[0].placeholder).toBe(
          'Ingresa un texto'
        )
      }
    )
  })

  describe('Error prop', () => {
    it(
      'Should recive error prop, render in component and set "warning"' +
        ' in className',
      () => {
        const { container, rerender, getByText } = render(
          <InputComponent
            valueId="input"
            value=""
            onChangeValue={mockOnChangeInput}
            type="text"
            label="Label"
          />
        )
        expect(container.getElementsByTagName('label')[0].className).toBe(
          'label'
        )
        rerender(
          <InputComponent
            valueId="input"
            value=""
            onChangeValue={mockOnChangeInput}
            type="text"
            error="error del input"
            label="Label"
          />
        )
        expect(container.getElementsByTagName('label')[0].className).toBe(
          'label warning'
        )
        expect(getByText('error del input')).toBeInTheDocument()
        expect(container.getElementsByTagName('input')[0].className).toBe(
          ' warning'
        )
      }
    )
  })

  describe('labelClassName prop', () => {
    it('Should recive labelClassName and replace label className', () => {
      const { container, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          label="Label"
        />
      )
      expect(container.getElementsByTagName('label')[0].className).toBe('label')
      rerender(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          label="Label"
          labelClassName="custom-className"
        />
      )
      expect(container.getElementsByTagName('label')[0].className).toBe(
        'custom-className'
      )
    })
  })

  describe('inputClassName prop', () => {
    it('Should recive inputClassName and replace input className', () => {
      const { container, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
        />
      )
      expect(container.getElementsByTagName('input')[0].className).toBe(' ')
      rerender(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          inputClassName="custom-inputClassName"
        />
      )
      expect(container.getElementsByTagName('input')[0].className).toBe(
        'custom-inputClassName '
      )
    })
  })

  describe('readOnly prop', () => {
    it(
      'Should recive readOnly and not change input value and add disabled ' +
        'className to label',
      () => {
        const mockUtil = new TestUtil()
        const { container, rerender } = render(
          <InputComponent
            valueId="input"
            value=""
            onChangeValue={mockUtil.mockedFunction}
            type="text"
            readOnly={true}
            label="Input"
          />
        )
        expect(container.getElementsByTagName('input')[0].value).toBe('')
        fireEvent.change(container.getElementsByTagName('input')[0], {
          target: {
            value: 'change',
          },
        })
        rerender(
          <InputComponent
            valueId="input"
            value={mockUtil.getValue()}
            onChangeValue={mockUtil.mockedFunction}
            type="text"
            inputClassName="custom-inputClassName"
            label="Input"
            readOnly={true}
          />
        )
        expect(container.getElementsByTagName('input')[0].value).toBe('')
        expect(container.getElementsByTagName('label')[0].className).toContain(
          'disable'
        )
      }
    )
  })

  describe('textArea prop', () => {
    it('Should recive textArea prop and render textArea', () => {
      const { container, rerender } = render(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
        />
      )
      expect(container.getElementsByTagName('input')).toHaveLength(1)
      rerender(
        <InputComponent
          valueId="input"
          value=""
          onChangeValue={mockOnChangeInput}
          type="text"
          inputClassName="custom-inputClassName"
          textArea={true}
        />
      )
      expect(container.getElementsByTagName('input')).toHaveLength(0)
      expect(container.getElementsByTagName('textarea')).toHaveLength(1)
    })
  })
})
