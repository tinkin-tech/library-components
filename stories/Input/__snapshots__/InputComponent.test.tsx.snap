// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent test Should take the snapshot with require props 1`] = `
<div
  className="input-component input p-t p-b flex-column"
  id="input-component"
>
  <div
    className="flex-space-between"
  >
    <label
      className="label "
      data-testid="label-component"
      htmlFor="input"
    >
      Label*
      Â 
    </label>
    
  </div>
  <input
    autoComplete="off"
    className=""
    data-testid="input-component"
    disabled={false}
    id="input"
    name="input"
    onChange={[Function]}
    placeholder="placeholder"
    spellCheck={false}
    type="text"
    value="value"
  />
  
</div>
`;
